### PR TITLE
Add mutex to pass "go test -race"

### PR DESCRIPTION
**Pull request by anfernee**
Add mutex to internal state so it's thread-safe (goroutine-safe) to access and monitor in different goroutines. Pass "go test -race".

### CODE REVIEW

**Comment by cheftako** [comment]
cc @caesarxuchao

File: pkg/agent/agentclient/client.go [hunk 1]
```diff
@@ -153,9 +155,13 @@ func (a *AgentClient) Serve(stopCh <-chan struct{}) {
 					}
 
 					close(dataCh)
+
+					a.connLock.Lock()
 					delete(a.connContext, connID)
+					a.connLock.Unlock()
 				},
 			}
+			a.connLock.Unlock()
```
Comment (cheftako) [inline pkg/agent/agentclient/client.go]: We need to get better about using defer.

File: .travis.yml [hunk 2]
```diff
@@ -10,4 +10,4 @@ before_script:
   - go install github.com/golangci/golangci-lint/cmd/golangci-lint
 script:
   - golangci-lint run --skip-files ".*.pb.go"
-  - go test -v ./... # go test -v -race ./... is failing, should be fixed
+  - go test -v -race ./...
```
Comment (Sh4d1) [inline .travis.yml]: Move the race test inside the makefile. The prow job (travis is not used) launches `make test`

File: pkg/agent/agentclient/client.go (hunk 1 above)
Comment (anfernee) [reply pkg/agent/agentclient/client.go]: the only way is to have a method for every state change. I hear you. Let me address it later.

File: pkg/agent/agentclient/client.go [hunk 3]
```diff
@@ -171,7 +177,11 @@ func (a *AgentClient) Serve(stopCh <-chan struct{}) {
 			klog.Infof("received DATA(id=%d)", data.ConnectID)
 			klog.Infof("[tracing] %v", data)
 
-			if ctx, ok := a.connContext[data.ConnectID]; ok {
+			a.connLock.RLock()
```
Comment (zanetworker) [inline pkg/agent/agentclient/client.go:180]: can we somehow reduce duplicate code by putting the wrapper code in a function? 
```suggestion
			a.wrapLocks(connContext)
```

**Comment by Jefftree** [comment]
This has been fixed by https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/106
/close

**Pull request closed without merging**